### PR TITLE
Stop running when a user scripts fails

### PR DESCRIPTION
**Pull request by chamini2**
TODO:
- [x] integration tests

### CODE REVIEW

File: src/fal/run_scripts.py [hunk 1]
```diff
@@ -86,6 +86,14 @@ def _prepare_exec_script(script: FalScript, project: FalProject) -> bool:
     return success
 
 
+def handle_run_results(scripts: List[FalScript], results: List[bool]):
```
Comment (mkam039) [inline src/fal/run_scripts.py:89]: Why don't we incorporate some of this into `run_scripts`? Maybe we can do mapping/filtering there and just raise an Error in a runner.

**Review by mkam039 — COMMENTED** [review]
Nice solution. Have a comment about implementation.

File: src/fal/run_scripts.py (hunk 1 above)
Comment (chamini2) [reply src/fal/run_scripts.py:89]: Yes, you are right. What if we also raise in the run_scripts for now? Just make it do everything about a script and later we change if we need to. What do you think?

Comment (mkam039) [reply src/fal/run_scripts.py:89]: Wouldn't this stop the entire fal run? Ideally we want to finish running everything we can and then raise an error.

Comment (chamini2) [reply src/fal/run_scripts.py:89]: I think we are OK like this for now, and we eventually will move to printing dbt-style logging like

```
Concurrency: 4 threads (target='dev')
1 of 2 START global script scripts/a.py
2 of 2 START model_a script scripts/a.py
2 of 2 OK model_a script scripts/a.py
1 of 2 OK global script scripts/a.py
```

Or something like that

File: src/fal/cli/flow_runner.py [hunk 2]
```diff
@@ -14,12 +14,16 @@ def fal_flow_run(parsed):
     node_graph = NodeGraph.from_fal_dbt(fal_dbt)
     execution_plan = ExecutionPlan.create_plan_from_graph(parsed, node_graph)
 
-    run_scripts(_id_to_fal_scripts(node_graph, execution_plan.before_scripts), project)
+    before_scripts = _id_to_fal_scripts(node_graph, execution_plan.before_scripts)
+    results = run_scripts(before_scripts, project)
+    handle_run_results(before_scripts, results)
```
Comment (turbo1912) [inline src/fal/cli/flow_runner.py:19]: let's call this something like `throw_if_failed`

otherwise I am good with this as discussed on Tuesday

**Review by turbo1912 — APPROVED** [review]

**Comment by turbo1912** [comment]
I approved but looks like tests are failing

**Review by turbo1912 — APPROVED** [review]

**Pull request merged**